### PR TITLE
Implement eig jvp

### LINKED ISSUE #2748: autodiff support for jax.numpy.linalg.eig
Note that `eigh` is already taken care of.

### PR DESCRIPTION
**Pull request by j-towns**
Currently draft status because the new test is failing due to incorrect gradient values. I haven't yet figured out the cause.

Fixes #2748.

TODO:
 - [ ] Fix reverse mode
 - [ ] Implement jvp for the left eigenvectors

### CODE REVIEW

**Comment by levskaya** [comment]
@j-towns is your gradient error showing up for real eigensystems or complex?  given that it's nonsymm `eig` could it be the latter?  i.e. are you doing a simple elementwise comparison of complex eigenvectors?  The phase freedom of complex entries makes numeric eigenvector gradient checking a bit more complicated than just using `check_grads`.

**Comment by j-towns** [comment]
Yeah indeed. I've dug into the implementation details of eig and it seems the 'phase freedom' is actually constrained by setting the largest component of each eigenvector to be real, thus the output of `(np|jnp).linalg.eig` is actually uniquely determined.

However it's questionable whether we should implement gradients that cater to that specific choice of constraint, since it's pretty arbitrary, and it feels like users shouldn't be depending on it in their code...

Some discussion on this towards the bottom of #2748. I'm not sure what the right answer is for these grads atm.
